inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -219,12 +208,29 @@ def article_exists(obj, eng):
     return False
 
 
+def is_being_harvested_on_legacy(record):
+    """Return True if the record is being harvested on Legacy.
+
+    If the record belongs to one of the CORE arXiv categories then it
+    is already being harvested on Legacy.
+    """
+    arxiv_categories = list(chain.from_iterable(
+        get_value(record, 'arxiv_eprints.categories', [])))
```

Comment:
Mmh. I am not sure this is more readable than the previous implementation... but OK.